presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java

```diff
@@ -1793,9 +1793,39 @@ public OptionalLong metadataDelete(ConnectorSession session, ConnectorTableHandl
     }
 
     @Override
-    public boolean supportsMetadataDelete(ConnectorSession session, ConnectorTableHandle tableHandle)
+    public boolean supportsMetadataDelete(ConnectorSession session, ConnectorTableHandle tableHandle, Optional<ConnectorTableLayoutHandle> tableLayoutHandle)
     {
-        return true;
+        if (!tableLayoutHandle.isPresent()) {
```

Comment:
do we need a `requireNonNull` check for the arguments here or are they handled in the caller?

Reply:
From what I've seen in Presto codebase, `requireNonNull` is used only in constructors.

Reply:
sure, the question was more geared towards do we need to null check using either `requireNonNull` or explicit `arg==null` .